src/Main.elm

```diff
@@ -0,0 +1,375 @@
+            in
+            { model
+                | boardSize = newBoardSize
+                , board = Dict.empty
+                , currentPlayer = X
+                , gameState = Started
+            }
+
+        ToggleModal ->
+            -- Hide the modal to change the board size.
+            { model | toggleModal = False }
+
+
+
+-- Name of Player
+
+
+writePlayer : Player -> String
+writePlayer player =
+    case player of
+        X ->
+            "Player X"
+
+        O ->
+            "Player O"
+
+
+
+-- Showing the symbol for each player
+
+
+showPlayer : Player -> String
+showPlayer player =
+    case player of
+        X ->
+            "X"
+
+        O ->
+            "O"
+
+
+
+-- Status of the game. Who's turn is it, whether a player has won, or there was a draw.
+
+
+viewStatus : GameState -> Player -> String
+viewStatus gameState currentPlayer =
+    case gameState of
+        Finished winner ->
+            case winner of
+                Just player ->
+                    writePlayer player ++ " Won!"
+
+                Nothing ->
+                    "Draw"
+
+        Started ->
+            "Now playing: " ++ writePlayer currentPlayer
+
+
+
+-- Board visualisation, while saving the position of each player (X, Y)
+
+
+viewBoard : Int -> Board -> List (Html Msg)
+viewBoard boardSize board =
+    List.range 0 (boardSize - 1)
+        |> List.map
+            (\x ->
+                div [ css [ rowStyle ] ]
+                    (List.range 0 (boardSize - 1)
+                        |> List.map
+                            (\y ->
+                                div [ onClick <| SquareClick ( x, y ), css [ squareStyle ] ]
+                                    [ case Dict.get ( x, y ) board of
+                                        Just player ->
+                                            text <| showPlayer player
+
+                                        Nothing ->
+                                            text <| ""
+                                    ]
+                            )
+                    )
+            )
+
+
+view : Model -> Html Msg
+view model =
+    div [ css [ mainStyle ] ]
+        [ if model.toggleModal then
+            modal model.boardSize ToggleModal DecCell AddCell
+
+          else
+            div [] []
+        , div [ css [ titleStyle ] ] [ text <| viewStatus model.gameState model.currentPlayer ]
+        , div [] <| viewBoard model.boardSize model.board
+        , case model.gameState of
+            Started ->
+                div [] []
+
+            Finished _ ->
+                button [ css [ playAgainButtonStyle ], onClick PlayAgain ] [ text "Play Again" ]
+        ]
+
+
+
+-- Reusable Function
+
+
+indices : Model -> List Int
+indices model =
+    List.range 0 (model.boardSize - 1)
+
+
+
+-- Function to get a row from the board
+
+
+row : Model -> b -> List ( Int, b )
+row model y =
+    indices model
+        |> List.map (\x -> ( x, y ))
```

Comment:
For me it seems weird that those functions take Model as an input. I'd suggest a change to just Int and re-naming of the args to be more explicit
```suggestion
row : Int -> Int -> List ( Int, Int )
row boardSize rowIndex =
    indices boardSize
        |> List.map (\columnIndex -> ( columnIndex, rowIndex ))
```